src/containers/passwordReset/ResetPassword.jsx

```diff
@@ -0,0 +1,96 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import ResetPassword from '../../components/passwordReset/ResetPassword';
+import { resetPasswordAction } from '../../store/actions/async/passwordResetActions';
+import { resetState } from '../../store/actions/sync/passwordResetActions';
+import '../../assets/styles/Signup.scss';
+
+export class PasswordResetContainer extends Component {
+
+    state = {
+        buttonDisabled: true,
+        password: "",
+        confirmPassword: "",
+        errorMsg : null
+    }
+
+    onInputChange = (e) => {
+        this.setState(
+            {buttonDisabled : !(this.state.password.length >= 7),
+            [e.target.name] : e.target.value}
+        )
+    }
+
+    onSubmit = () => {
+        if (this.state.password !== this.state.confirmPassword){
+            this.setState({errorMsg: "Passwords do not match!"})
+        }
+        else {
+            this.setState({errorMsg: null});
+            let token = this.props.match.params.token
+            const { resetPassword } = this.props;
+            resetPassword(token, this.state.password)
+        }
+    };
+
+    render() {
+
+        let showError = false
+        let incomingError = this.props.error
+
+        if (this.state.errorMsg !== null) {
+            showError = true
+        }else if (incomingError !== null) {
+            this.setState({errorMsg: (('user' in incomingError)? incomingError.user.message :
+            incomingError.errors.error[0])})
+            showError = true
+        };
+
+        return (
+            <div className="Signup">
+                <ResetPassword
+                    disabled={this.state.buttonDisabled}
```

Comment:
Make use of destructuring to prevent duplication of `this.state` and `this.props`